### DISCUSSION TITLE
OpenAPI sorts examples alphabetically

### DISCUSSION THREAD

**Discussion by andrey-kozel** [fastapi/fastapi#13028] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
from enum import Enum

from pydantic import BaseModel, Field


class UserDefiPositionEnum(str, Enum):
    LIQUID_STACKING_TOKEN = 'liquid_staking_token'
    TOKEN = 'token'

class UserTokenBalance(BaseModel):
    type: UserDefiPositionEnum = Field(description="", examples=["supply"])
    id: str = Field(description="", examples=["aave_v2_eeth"])
    chain_id: str = Field(description="", examples=["ethereum"])
    chain_name: str = Field(description="", examples=["Ethereum"])
    value_usd: float = Field(description="", examples=[3123])
    value_usd_change_1d: float = Field(description='', examples=[3123])
    attributes: dict = Field(description='')


class UserBalancesData(BaseModel):
    assets: list[UserTokenBalance] = Field(description='', examples=[[
        {
            "type": "token",
            "id": "eth",
            "defi_id": None,
            "defi_name": None,
            "chain_id": "ethereum-mainnet",
            "chain_name": "Ethereum mainnet",
            "value_usd": 0.55,
            "value_usd_change_1d": 0.02,
            "attributes": {
                "token_id": "eth",
                "token_symbol": "ETH",
                "contract_address": None,
                "decimals": 18,
                "format": "native",
                "amount": 189721692804000,
                "amount_change_1d": 0
            }
        }
    ]])


class UserBalancesV2(BaseModel):
    data: UserBalancesData = Field(description='')
```

### Description

I'm adding custom examples section to my response object and expecting to see it in the same format on swagger, but all the fields are sorted in alphabetical order.
<img width="1296" alt="image" src="https://github.com/user-attachments/assets/e57f8c0f-847f-45de-a35c-91bc37d968fc">
 

### Operating System

macOS

### FastAPI Version

fastapi = "0.115.0"

### Pydantic Version

2.10.2

### Python Version

 3.12.7

### Additional Context

As far as I see, pydantic had an issue with sorting, but it exposed sorting method in the schema generator (https://github.com/pydantic/pydantic/issues/7580)

In the `openapi/utils.py` generator is hardcoded
<img width="731" alt="image" src="https://github.com/user-attachments/assets/3f33c9f9-56bf-4bb6-bbbf-6445d828e0f9">

Is it possible, to add ability to configure generator? I would like to use my own with overridden sort method

**Comment by valentinDruzhinin** [comment]
If you really need the order you can define the example as string:
```
class UserBalancesData(BaseModel):
    assets: list[UserTokenBalance] = Field(description='', example="""{
    "type": "token",
    "id": "eth",
    "defi_id": null,
    "defi_name": null,
    "chain_id": "ethereum-mainnet",
    "chain_name": "Ethereum mainnet",
    "value_usd": 0.55,
    "value_usd_change_1d": 0.02,
    "attributes": {
        "token_id": "eth",
        "token_symbol": "ETH",
        "contract_address": null,
        "decimals": 18,
        "format": "native",
        "amount": 189721692804000,
        "amount_change_1d": 0
    }
}""")
```

In that case, the order will be the same as you define.

**Comment by yakubka** [comment]
The alphabetical ordering of examples comes from how Python dicts are serialized to JSON in the OpenAPI schema generation. When FastAPI collects multiple `examples` from `openapi_extra` or from `Body(examples=...)`, it does not enforce a specific order, and the JSON serializer (which uses a standard dict) preserves insertion order in Python 3.7+. If your examples are ending up alphabetical, they are likely being defined in a dict literal that is alphabetically ordered, or they are being merged in a way that sorts them.

To control the order of examples, define them as a list of `(name, value)` pairs and convert to a dict at the end.

```python
from fastapi import Body
from typing import Annotated

@app.post("/items")
async def create_item(
    item: Annotated[
        Item,
        Body(
            examples={
                "first_example": {"summary": "First", "value": {"name": "A"}},
                "second_example": {"summary": "Second", "value": {"name": "B"}},
            }
        ),
    ]
):
    ...
```

Python dict literals preserve insertion order, so the order you write them is the order they appear in the spec and in Swagger UI. If the order is still wrong, check whether any middleware or post-processing step is re-sorting the schema dict.
